src/services/main_service.ts

```diff
@@ -28,12 +28,12 @@ export class MainService extends PluginServiceClass {
         .getCharacteristic(this.hap.Characteristic.RotationSpeed)
         .onGet(() => this.getSpeed())
         .onSet((newState) => this.setSpeed(newState));
+      this.service.setPrimaryService;
     } else {
       this.service = new this.hap.Service.Switch(this.config.name, "Vacuum");
+      this.service.setPrimaryService;
     }
 
-    if (this.service.setPrimaryService) this.service.setPrimaryService(true);
```

Comment:
In some versions of homebridge, `setPrimaryService` is not available, that's why I had to add that validation in the past.

If the method exists, call it.